operator-common/src/main/java/io/strimzi/operator/common/featuregates/FeatureGates.java

```diff
@@ -148,15 +162,16 @@ public boolean equals(Object o) {
             return false;
         } else {
             FeatureGates other = (FeatureGates) o;
-            // return Objects.equals(dummyFeatureGate, other.dummyFeatureGate);
-            return Objects.equals(serverSideApplyPhase1, other.serverSideApplyPhase1);
+            // return Objects.equals(dummyFeatureGate, other.dummyFeatureGate)
+            return Objects.equals(serverSideApplyPhase1, other.serverSideApplyPhase1)
+                && Objects.equals(useConnectBuildWithBuildah, other.useConnectBuildWithBuildah);
         }
     }
 
     @Override
     public int hashCode() {
-        // return Objects.hashCode(dummyFeatureGate);
-        return Objects.hashCode(serverSideApplyPhase1);
+        // return Objects.hashCode(dummyFeatureGate)
+        return Objects.hashCode(serverSideApplyPhase1) + Objects.hashCode(useConnectBuildWithBuildah);
```

Comment:
I don't think this is a good hash code. You would for example do something like ...

```
    result = serverSideApplyPhase1.hashCode();
    result = 31 * result + useConnectBuildWithBuildah.hashCode();
    return result;
```

or something liek that. IDE should know how to generate it.